exercises/Robert_Applin/ex01_basics/src/main.cpp

```diff
@@ -0,0 +1,203 @@
+/**
+ * Skeleton main routine
+ */
+
+#include <algorithm>
+#include <cctype>
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <iterator>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+using WordCount = std::vector<std::pair<std::string, std::size_t>>;
+
+/**
+ * @return An iterator pointing to the next letter contained
+ *         in the supplied set.
+ */
+std::string::const_iterator
+findNextCharacter(std::string::const_iterator fromIterator,
+                  std::string::const_iterator toIterator,
+                  const std::string &characters) {
+  for (auto it = fromIterator; it < toIterator; ++it) {
```

Comment:
Perhaps use `const auto` for clearer intent?

Reply:
Also update other places where this is true.

Reply:
Good point, but it will be changing, so it can't be const. Is that right?

Reply:
Ah apologies. I think you're correct. The iterator needs to be mutable here. Forget this comment.